sktime/forecasting/base/adapters/_statsforecast.py

```diff
@@ -226,47 +220,35 @@ def _predict_interval(self, fh, X=None, coverage=0.90):
                 second level coverage fractions for which intervals were computed.
                     in the same order as in input `coverage`.
                 Third level is string "lower" or "upper", for lower/upper interval end.
-            Row index is fh. Entries are forecasts of lower/upper interval end,
+            Row index is fh, with additional (upper) levels equal to instance levels,
+                from y seen in fit, if y_inner_mtype is Panel or Hierarchical.
+            Entries are forecasts of lower/upper interval end,
                 for var in col index, at nominal coverage in second col index,
                 lower/upper depending on third col index, for the row index.
                 Upper/lower interval end forecasts are equivalent to
                 quantile forecasts at alpha = 0.5 - c/2, 0.5 + c/2 for c in coverage.
         """
-        # initializaing cutoff and fh related info
-        cutoff = self.cutoff
-        fh_oos = fh.to_out_of_sample(cutoff)
-        fh_ins = fh.to_in_sample(cutoff)
-        fh_is_in_sample = fh.is_all_in_sample(cutoff)
-        fh_is_oosample = fh.is_all_out_of_sample(cutoff)
-
-        # prepare the return DataFrame - empty with correct cols
-        var_names = ["Coverage"]
-        int_idx = pd.MultiIndex.from_product([var_names, coverage, ["lower", "upper"]])
-        pred_int = pd.DataFrame(columns=int_idx)
-
-        kwargs = {"X": X, "return_pred_int": True, "alpha": coverage}
-        # all values are out-of-sample
-        if fh_is_oosample:
-            _, y_pred_int = self._predict_fixed_cutoff(fh_oos, **kwargs)
-
-        # all values are in-sample
-        elif fh_is_in_sample:
-            _, y_pred_int = self._predict_in_sample(fh_ins, **kwargs)
-
-        # if all in-sample/out-of-sample, we put y_pred_int in the required format
-        if fh_is_in_sample or fh_is_oosample:
-            # needs to be replaced, also seems duplicative, identical to part A
-            for intervals, a in zip(y_pred_int, coverage):
-                pred_int[("Coverage", a, "lower")] = intervals["lower"]
-                pred_int[("Coverage", a, "upper")] = intervals["upper"]
-            return pred_int
-
-        # both in-sample and out-of-sample values (we reach this line only then)
-        # in this case, we additionally need to concat in and out-of-sample returns
-        _, y_ins_pred_int = self._predict_in_sample(fh_ins, **kwargs)
-        _, y_oos_pred_int = self._predict_fixed_cutoff(fh_oos, **kwargs)
-        for ins_int, oos_int, a in zip(y_ins_pred_int, y_oos_pred_int, coverage):
-            pred_int[("Coverage", a, "lower")] = pd.concat([ins_int, oos_int])["lower"]
-            pred_int[("Coverage", a, "upper")] = pd.concat([ins_int, oos_int])["upper"]
-
-        return pred_int
+        X_predict_input = X if X is None else X.to_numpy(copy=False)
```

Comment:
it would be slightly less obscure if you say: `if X is not None: X_predict_input = X.to_numpy, else: X_predict_input = None`